src/layer/AbstractLayer.ts

```diff
@@ -60,21 +58,103 @@ export class AbstractLayer {
     if (!this.dataset || !this.dataset.orbitTimeMinutes) {
       throw new Error('Orbit time is needed for grouping tiles into flyovers.');
     }
-
     if (bbox.crs !== CRS_EPSG4326) {
       throw new Error('Currently, only EPSG:4326 in findFlyovers');
     }
 
-    let tiles: Tile[] = [];
+    const orbitTimeMS = this.dataset.orbitTimeMinutes * 60 * 1000;
+    const bboxGeometry: Geom = this.roundCoordinates([
+      [
+        [bbox.minX, bbox.minY],
+        [bbox.maxX, bbox.minY],
+        [bbox.maxX, bbox.maxY],
+        [bbox.minX, bbox.maxY],
+        [bbox.minX, bbox.minY],
+      ],
+    ]);
+    const bboxArea = area({
+      type: 'Polygon',
+      coordinates: bboxGeometry,
+    });
+
+    let flyovers: FlyoverInterval[] = [];
+    let flyoverIndex = 0;
+    let currentFlyoverGeometry: Geom = null;
+    let nTilesInFlyover;
+    let sumCloudCoverPercent;
     for (let i = 0; i < maxFindTilesRequests; i++) {
-      const { tiles: partialTiles, hasMore } = await this.findTiles(
+      // grab new batch of tiles:
+      const { tiles, hasMore } = await this.findTiles(
         bbox,
         fromTime,
         toTime,
         tilesPerRequest,
         i * tilesPerRequest,
       );
-      tiles = tiles.concat(partialTiles);
+
+      // apply each tile to the flyover to calculate coverage:
+      for (let tileIndex = 0; tileIndex < tiles.length; tileIndex++) {
+        // first tile ever? just add its info and continue:
+        if (flyovers.length === 0) {
+          flyovers[flyoverIndex] = {
+            fromTime: tiles[tileIndex].sensingTime,
+            toTime: tiles[tileIndex].sensingTime,
+            coveragePercent: 0,
+            meta: {},
+          };
+          currentFlyoverGeometry = tiles[tileIndex].geometry.coordinates as Geom;
+          sumCloudCoverPercent = tiles[tileIndex].meta.cloudCoverPercent;
+          nTilesInFlyover = 1;
+          continue;
+        }
+
+        // append the tile to flyovers:
+        const prevDateMS = flyovers[flyoverIndex].fromTime.getTime();
+        const currDateMS = tiles[tileIndex].sensingTime.getTime();
+        const diffMS = Math.abs(prevDateMS - currDateMS);
+        if (diffMS > orbitTimeMS) {
+          // finish the old flyover:
+          try {
+            flyovers[flyoverIndex].coveragePercent = this.calculateCoveragePercent(
+              bboxGeometry,
+              bboxArea,
+              currentFlyoverGeometry,
+            );
+          } catch (err) {
+            // if anything goes wrong, play it safe and set coveragePercent to null:
+            flyovers[flyoverIndex].coveragePercent = null;
+          }
+          if (sumCloudCoverPercent !== undefined) {
+            flyovers[flyoverIndex].meta.averageCloudCoverPercent = sumCloudCoverPercent / nTilesInFlyover;
```

Comment:
As discussed in person, `coveragePercent` should also be taken into account, but I think for now this is good enough.